spec/src/main/asciidoc/configexamples.asciidoc

```diff
@@ -129,3 +129,155 @@ public class InjectedConfigUsageSample {
 }
 ----
 
+=== Aggregate related properties into a single POJO Example
+
+When injecting a number of related configuration properties, it can be tedious to repeat the statement of `ConfigProperty` in scatter places. 
+Since they are related, it makes more sense to aggregate them into a single property class. 
+
+MicroProfile Config provides a way to look up a number of configuration properties starting with the same prefix using the `@ConfigProperties` annotation, e.g. `ConfigProperties(prefix="myPrefix")`.
+When annotating a class with `@ConfigProperties` or `@ConfigProperties(prefix="myPrefix")`, any of its fields, regardless of the visibility, maps to a configuration property via the following mapping rules. 
+
+* If the `prefix` is present, the field `x` maps to the configuration property `<prefix>.x`. 
+* If the `prefix` is absent, the field `x` maps to the property name `x`. 
+
+If the field name `x` needs to be different from the config property name `y`, use `@ConfigProperty(name="y")` to perform the transformation. 
+If the prefix is present, the field `x` maps to the configuration property `<prefix>.y`, otherwise `y`.
```

Comment:
I think we need to define a few more mapping rules. For instance, a dash `-`, is a valid char to be part of the name of a configuration property. How do you map them to method names? Or should we consider that there is no convention to map these and you alway have to rely on the annotation to set the name?

Reply:
I am confused, isn't the mapping the other way around? The field name maps to a property, and as you can use all valid field names as property names this is fine. Or if you annotate the field and provide a string for name name the name can be any string which should make any valid property name work.

Reply:
It is tricky for the mapping rules as when we might have to look up client.name, client_name, client$name based on some mapping rules. Which one to match first? It will be very messy. Since the config property always rely on names to specify property names, we use the same approach. I suddently thought of something in this area. Should we force prefix to contain all characters? At the moment, we added dot between prefix and the rest part. I think it is better not to add dot as there might be other separator such as _ used in the config properties such as client_name, client_age.
Thoughts? @radcortez @jbee If you agree, I will update the last bit.

Reply:
I am happy either way. Implicit dot is more constrained but maybe a bit more intuitive, explicit is more flexible but also a bit less intuitive to me.

Reply:
I think it is nice to make prefix as the prefix without adding arbitratry separator. I have done the changes accordingly. Please double check again as I aim to merge this PR asap.

Reply:
Not sure if I agree with this. It may cause additional issues if we want to support nested / subnamespaces.

Still, this doesn't address the original comment. For instance if I have the following property:

```
foo-bar=baz
```

There is no way I can express that in a method name, unless I use the annotation to set the name. For instance Spring uses these rules for binding: https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-external-config-relaxed-binding

Meaning that a method with the name `fooBar` could bind to the property `fooBar` or `foo-bar`, or `foo_bar` or `FOOBAR`.

Reply:
I think for the prefix it allows for all characters. We don't need to transform it and keep it as exact. I prefer to use the exact prefix without adding the additional dot as the prefix might be ending with $ or - or _. 

I was thinking not to transform the field but using `ConfigProperty(name="") to specify the exact name. As for environment variable mapping, we have defined the rules already, which is to transforming all illegal characters to _ and then try to search all uppercase.

However, if we want to support mapping. @radcortez I think one of the possible tranformation is to translate all illegal character to _

`foo-bar=baz`
matches to the field `foo_bar`, which is consistent with MP Config mapping rule.

Reply:
It is fine by me, I just wanted to add the clarification saying that we only support exact match or `@ConfigProperty` name, so people don't expect any kind of mapping rules.

Reply:
ok thanks @radcortez ! I'll merge my PR now.